### PR TITLE
MVP fertility part 2

### PR DESCRIPTION
**Pull request by LoraArmstrong**
- Project births from fixed rates
- Remove death-rates and birth-rates from the mortality and fertility modules outputs.
- Calculate historic Age-Specific Fertility Rates

### CODE REVIEW

**Comment by acron0** [comment]
Because the test data is inside `resources` it's getting packaged into the library which isn't what we want. You can create a new directory (`dev-resources` for example), put the test files there and use the `:dev` profile to add it to your list of resources for `lein test`

**Comment by acron0** [comment]
Because you're using resources in this manner, you can don't need to refer to the full path of the file either, I believe:

https://github.com/MastodonC/witan.models.demography/blob/aabd106416f0290077a97e9911af637de5513b39/src/witan/models/load_data.clj#L20

I think `(io/as-file ...)` should be `(io/file (io/resource ...))` but might need some experimentation. Either way, this is file to roll :shipit:

File: src/witan/models/dem/ccm/schemas.clj [hunk 1]
```diff
@@ -35,13 +35,17 @@
...
+                           [:year s/Int] [:births double]]))
+
 ;;For core CCM projection loop using fert/mort/mig inputs from files
 (def BirthsBySexSchema
   (make-ordered-ds-schema [[:gss-code s/Str] [:sex s/Str] [:births double]]))
 
 (def DeathsOutputSchema
   (make-ordered-ds-schema [[:gss-code s/Str] [:sex s/Str] [:age s/Int] [:year s/Int]
-                           [:death-rate double] [:popn double] [:deaths double]]))
+                           [:popn double] [:deaths double]]))
```
Comment (sunnytownsend) [inline src/witan/models/dem/ccm/schemas.clj]: remove popn too - unless good reason I'm missing

File: src/witan/models/dem/ccm/fert/fertility_mvp.clj [hunk 2]
```diff
@@ -4,8 +4,76 @@
             [clojure.core.matrix.dataset :as ds]
             [witan.datasets :as wds]
             [witan.models.dem.ccm.schemas :refer :all]
+            [incanter.core :as i]
             [schema.core :as s]))
 
+(defn create-popn-at-risk-birth
+  "Creates a population at risk for calculating historic age specific
```
Comment (sunnytownsend) [inline src/witan/models/dem/ccm/fert/fertility_mvp.clj]: change 'historic' to 'base'

File: src/witan/models/dem/ccm/fert/fertility_mvp.clj [hunk 3]
```diff
@@ -4,8 +4,76 @@
...
+      (wds/join popn-at-risk [:gss-code :sex :age])
+      (wds/add-derived-column :births [:popn :base-fert-rate]
+                              (fn [p b] (* ^double p b)))
+      (wds/add-derived-column :year [:year] inc)
+      (wds/rollup :sum :births [:gss-code :year])
+      (ds/rename-columns {:births :estimated-births})))
+
+(defn calc-actual-births
+  "Takes in the historic data of total births, filters for
+  the base year and groups the number of births for that year
```
Comment (sunnytownsend) [inline src/witan/models/dem/ccm/fert/fertility_mvp.clj]: change 'groups' to 'totals'

File: src/witan/models/dem/ccm/fert/fertility_mvp.clj [hunk 4]
```diff
@@ -4,8 +4,76 @@
...
+  "Creates a population at risk for calculating historic age specific
+  fertility rates. Takes in the historic population and the base year
+  for the fertility projection."
+  [historic-population fert-last-yr]
+  (i/query-dataset historic-population
+                   {:year (dec fert-last-yr)}))
+
+(defn calc-estimated-births
+  "Takes in the population at risk and the dataset for base age specific
+  fertility rates data to calculate the estimated births."
```
Comment (sunnytownsend) [inline src/witan/models/dem/ccm/fert/fertility_mvp.clj]: add at end: which is totalled by gss-code and year

Comment (sunnytownsend) [inline src/witan/models/dem/ccm/fert/fertility_mvp.clj]: lgtm once small changes addressed I am happy to :shipit:

**Pull request merged**